File: SmartPhone++/SmartPhone.Services/Database/Part.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Services.Database
{
    public class Part
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(1000)]
        public string? Description { get; set; }
        
        [Required]
        [Range(0, double.MaxValue)]
        public decimal Price { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal? CostPrice { get; set; }
        
        [Required]
        [Range(0, int.MaxValue)]
        public int StockQuantity { get; set; }
        
        [Range(0, int.MaxValue)]
        public int? MinimumStockLevel { get; set; }
        
        public string? SKU { get; set; }
        
        public string? PartNumber { get; set; }
        
        public string? Brand { get; set; }
        
        public string? Model { get; set; }
        
        public string? Color { get; set; }
        
        [MaxLength(50)]
        public string? Condition { get; set; } // New, Used, Refurbished, OEM, Aftermarket
        
        [MaxLength(50)]
        public string? Grade { get; set; } // A, B, C, D
        
        public bool IsActive { get; set; } = true;
        
        public bool IsOEM { get; set; } = false;
        
        public bool IsCompatible { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime? UpdatedAt { get; set; }
        
        // Foreign keys
        public int PartCategoryId { get; set; }
        
        // Navigation properties
        public PartCategory PartCategory { get; set; } = null!;
        public ICollection<ServicePart> ServiceParts { get; set; } = new List<ServicePart>();
        public ICollection<PartCompatibility> CompatiblePhones { get; set; } = new List<PartCompatibility>();
    }
} 

File: SmartPhone++/UI/smartphone_desktop_admin/lib/model/part.dart
import 'package:json_annotation/json_annotation.dart';

part 'part.g.dart';

@JsonSerializable()
class Part {
  final int id;
  final String name;
  final String? description;
  final double price;
  final double? costPrice;
  final int stockQuantity;
  final int? minimumStockLevel;
  final String? sku;
  final String? partNumber;
  final String? brand;
  final String? model;
  final String? color;
  final String? condition;
  final String? grade;
  final bool isActive;
  final bool isOEM;
  final bool isCompatible;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final int partCategoryId;
  final String partCategoryName;

  Part({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    this.costPrice,
    required this.stockQuantity,
    this.minimumStockLevel,
    this.sku,
    this.partNumber,
    this.brand,
    this.model,
    this.color,
    this.condition,
    this.grade,
    required this.isActive,
    required this.isOEM,
    required this.isCompatible,
    required this.createdAt,
    this.updatedAt,
    required this.partCategoryId,
    required this.partCategoryName,
  });

  factory Part.fromJson(Map<String, dynamic> json) => _$PartFromJson(json);
  Map<String, dynamic> toJson() => _$PartToJson(this);
} 